MeesGame/Gameplay/UIObjects/Objects/Button.cs

```diff
@@ -79,17 +74,26 @@ public override void DrawTask(GameTime gameTime, SpriteBatch spriteBatch)
         public override void Update(GameTime gameTime)
         {
             base.Update(gameTime);
-            Invalidate = true;
         }
 
-        public override void HandleInput(InputHelper inputHelper)
+        public bool Selected
         {
-            base.HandleInput(inputHelper);
-            if (Clicked)
+            get { return selected; }
+            set { selected = value; }
+        }
+
+
+        public override bool Invalidate
+        {
+            get
             {
-                OnClick?.Invoke(this);
+                return base.Invalidate;
             }
 
+            set
+            {
+                base.Invalidate = true;
```

Comment:
What is this????? Are you sure Invalidate = false should set Invalidate to true?